components/Button/SwitchNetwork.tsx

```diff
@@ -0,0 +1,49 @@
+import { Button, ButtonProps, useToast } from '@chakra-ui/react'
+import { formatError } from '@nft/hooks'
+import { PropsWithChildren, useCallback } from 'react'
+import { useChainId, useSwitchNetwork } from 'wagmi'
+
+const ButtonWithNetworkSwitch = ({
+  children,
+  chainId,
+  ...props
+}: PropsWithChildren<
+  ButtonProps & {
+    chainId: number
+  }
+>): JSX.Element => {
+  const currentChainId = useChainId()
+  const { switchNetwork, isLoading } = useSwitchNetwork({
+    chainId,
+    throwForSwitchChainNotSupported: true,
+    onError(error) {
+      toast({
+        title: formatError(error),
+        status: 'error',
+      })
+    },
+  })
+  const toast = useToast()
+
+  const handleSwitchNetwork = useCallback(() => {
+    if (!switchNetwork) return
+    switchNetwork()
```

Comment:
It doesn't work if you don't pass the chainId:
```suggestion
    switchNetwork(chainId)
```
I already fixed it in a previous commit. Any reason to revert it?

Reply:
I added it to the `switchNetwork` params

Reply:
Ok you moved it to a prop of `useSwitchNetwork` directly. It's working as expected